### ISSUE TITLE
Way to whitelist/override an entry in an enabled host file?

### ISSUE THREAD

**Issue by CollinChaffin** [gorhill/uMatrix#770]
Sorry but I saw a closed issue where @gorhill stated a while back that you could do a hidden dynamic doc allow, but I find no way to add a domain that appears in one or more enabled hosts files as an ALLOW override.  Can it be done and if not, why?

The only workaround I have found is to have to disable half the presented hosts files and all their protection (and kill most of UM protection), just because a few domains absolutely need to be allowed so it would be really nice to have a way to perform an override and hopefully I'm just not seeing it.

Thanks for the help!

Collin

**Comment by gorhill** [comment]
Did you try to just click on the rule in the matrix?

**Comment by CollinChaffin** [comment]
@gorhill Why did you close this so quickly?  You are awfully short with your loyal users of your stuff I'm sorry but the docs aren't clear and your answer makes no sense so can you explain?  Let me re-phrase the question - when I enable ALL host files lists - I want xyz.com allowed.  EVERY visit to anything on xyz.com gives me a page saying screw you - denied.  Unlike ublock with an override - I cannot continue.  I go back and add rules - they don't work and the docs confirm HOST FILES are global and TRUMP everything.

Let's say xyz.com is literally found in ALL the lists.

So again, if xyz.com is in ALL THE HOST FILE LISTS, how can I allow xyz.com without deselecting all the lists and basically turning off umatix all together or is this not possible in your app?

**Comment by gorhill** [comment]
Nowhere does the doc say that rules from hosts files "trump everything", [it says](https://github.com/gorhill/uMatrix/wiki/Changes-from-HTTP-Switchboard#ubiquitous-rules-tab-replaced-by-hosts-files-tab):

> All hostnames in selected hosts files are interpreted as blacklisted hostnames in the global scope, so they propagate to narrower scopes just like in HTTPSB.

My comment above:

> Did you try to just click on the rule in the matrix -- like you would do for any rule you created yourself?

Did you? There is no special trick with document blocked by hosts file rules, just like any other un-breaking operation, you open the matrix panel, click on the rule which you want to change, force reload to see if it works, then persist if you like the result.

**Comment by jjohns71** [comment]
@CollinChaffin [Please see my post](https://github.com/gorhill/uMatrix/issues/685#issuecomment-268620190) on #685   that details how to create and save global rules - I think it may be helpful for you. 

In the case of overriding a blocked domain that is from a hosts file. You can do this from the uMatrix panel while you are still on the **uMatrix has prevented the following page from loading** landing page.

Steps below:
1. Open the uMatrix panel, and click on the red highlighted (blocked) domain name under the 'All' section heading of the uMatrix panel. (The domain should now be highlighted in green)
2. [Save the new rule as described](https://github.com/gorhill/uMatrix/issues/685#issuecomment-268620190) in #685.

**Comment by gorhill** [comment]
You don't have to create a global rule for this case, `example.com example.com * allow` will work just fine -- assuming `example.com` was listed in one of the hosts files.

**Comment by jjohns71** [comment]
@gorhill That is correct and I should have made that more clear in my post, I just referenced the previous post because it explains how to make a rule permanent that I think a lot of users don't understand. 

If I typed up similar rules for saving rules for one domain, would you be willing to add the walk-through for saving/creating both global and domain bound rules to the Wiki?

**Comment by gorhill** [comment]
I made the wiki available to edit by all, just go ahead and add what you want.

**Comment by jjohns71** [comment]
Thanks @gorhill I created a new [wiki page](https://github.com/gorhill/uMatrix/wiki/Setting-Domain-Specific-&-Global-Rules) for demonstration of how to create global and domain specific rules... hopefully this is helpful.

**Comment by gorhill** [comment]
Thanks, I've added your page to a special entry on the wiki home page. Given that there are many ways to use uMatrix, I just put a warning that I may or may not agree on whether this is an optimal way of using uMatrix. Regardless, these are welcome as I just do not have the time to create full-blown doc.
